tests/test_mapping.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import torch
+
+
+class TestGetPeftModel:
+    RELOAD_WARNING_EXPECTED_MATCH = r"You are trying to modify a model .*"
+
+    @pytest.fixture
+    def get_peft_model(self):
+        from peft import get_peft_model
+
+        return get_peft_model
+
+    @pytest.fixture
+    def lora_config(self):
+        from peft import LoraConfig
+
+        return LoraConfig(target_modules="0")
+
+    @pytest.fixture
+    def base_model(self):
+        return torch.nn.Sequential(torch.nn.Linear(10, 2))
+
+    def test_get_peft_model_warns_when_reloading_model(self, get_peft_model, lora_config, base_model):
+        get_peft_model(base_model, lora_config)
+
+        with pytest.warns(UserWarning, match=self.RELOAD_WARNING_EXPECTED_MATCH):
+            get_peft_model(base_model, lora_config)
+
+    def test_get_peft_model_proposed_fix_in_warning_helps(self, get_peft_model, lora_config, base_model, recwarn):
+        peft_model = get_peft_model(base_model, lora_config)
+        peft_model.unload()
+        get_peft_model(base_model, lora_config)
+
+        warning_checker = pytest.warns(UserWarning, match=self.RELOAD_WARNING_EXPECTED_MATCH)
+
+        for warning in recwarn:
+            if warning_checker.matches(warning):
+                pytest.fail("Warning raised even though model was unloaded.")
```

Comment:
Maybe a bit of an edge case, but currently we would not detect if a user tries to create, say, a LoRA model based on a base model that was already modified with a prompt learning method, like prefix tuning. This is because those methods don't add any `BaseTunerLayer`s (which is what `get_layer_status` relies on). Implementing such a check is probably not easy and out of scope for this PR. We could still add an xfail-ing test though.

Reply:
Honestly, I'm not quite sure if it would just work or not. Coincidentally, there is a recent issue asking a similar question (#2307).

Reply:
For our sanity I'd suggest to go forward with the current state of things and once we know more about the interplay between soft-prompting vs. soft-prompting and soft-prompting vs. lora we adapt tests and/or implementation. WDYT?